File: deno/lib/oss/oss.service.ts
import * as ossDao from "./oss.dao.ts";

/**
 * 上传文件
 * @param {File} file
 */
export async function upload(
  file: File,
  opt?: {
    once?: number;
    is_public?: boolean;
    tenant_id?: string;
    db?: string;
    id?: string;
  },
) {
  const result = await ossDao.upload(file, opt);
  return result;
}

export async function statObject(id: string) {
  return await ossDao.statObject(id);
}

export async function getObject(id: string) {
  return await ossDao.getObject(id);
}

export async function deleteObject(id: string) {
  return await ossDao.deleteObject(id);
}

File: rust/src/common/oss/oss_service.rs
use color_eyre::eyre::Result;

use super::oss_dao::{self, StatObject};
use s3::request::{ResponseData, ResponseDataStream};

use crate::r#gen::base::tenant::tenant_model::TenantId;

/**
 * 上传文件
 */
#[allow(clippy::too_many_arguments)]
pub async fn put_object<S: AsRef<str>>(
  path: S,
  content: &[u8],
  content_type: &str,
  filename: &str,
  is_public: Option<&str>,
  tenant_id: Option<TenantId>,
  db: Option<&str>,
  id: Option<&str>,
) -> Result<ResponseData> {
  let res = oss_dao::put_object(
    path, content, content_type, filename,
    is_public, tenant_id, db, id,
  ).await?;
  Ok(res)
}

pub async fn head_object(
  path: &str,
) -> Result<Option<StatObject>> {
  let res = oss_dao::head_object(path).await?;
  Ok(res)
}

pub async fn get_object(
  path: &str,
) -> Result<Option<ResponseData>> {
  let res: Option<ResponseData> = oss_dao::get_object(path).await?;
  Ok(res)
}

#[allow(dead_code)]
pub async fn get_object_stream(
  path: &str,
) -> Result<Option<ResponseDataStream>> {
  let res: Option<ResponseDataStream> = oss_dao::get_object_stream(path).await?;
  Ok(res)
}

pub async fn delete_object(
  path: &str,
) -> Result<bool> {
  let res = oss_dao::delete_object(path).await?;
  Ok(res)
}
